app/scripts/directives/overview/serviceInstanceRow.js

```diff
@@ -39,10 +39,30 @@
       return _.get(row, ['state','serviceClasses', serviceClassName, 'description']);
     };
 
+    var updateProvisionStatus = function() {
+      var conditions = _.get(row.apiObject, 'status.conditions');
+      var readyCondition = _.find(conditions, {type: "Ready"});
+
+      row.provisionError = _.find(conditions, {type: "Failed"});
+
+      if (_.get(row.apiObject, 'metadata.deletionTimestamp')) {
+        row.provisionStatus = 'deleted';
+      } else if (row.provisionError) {
+        row.provisionStatus = 'error';
+      } else if (readyCondition && readyCondition.status === 'True') {
+        row.provisionStatus = 'ok';
+      } else {
+        row.provisionStatus = 'inProgress';
```

Comment:
Suggest `row.instanceStatus` with `'deleted'`, `'failed'`, `'ready'`, and `'pending'` values. I feel like it aligns a little better with the real API conditions.